File: db4on/core/src/core/com/db4o/QESmaller.cs
namespace com.db4o
{
	/// <exclude></exclude>
	public class QESmaller : com.db4o.QEAbstract
	{
		internal override bool Evaluate(com.db4o.QConObject a_constraint, com.db4o.QCandidate
			 a_candidate, object a_value)
		{
			if (a_value == null)
			{
				return false;
			}
			return a_constraint.GetComparator(a_candidate).IsSmaller(a_value);
		}

		public override void IndexBitMap(bool[] bits)
		{
			bits[com.db4o.QE.SMALLER] = true;
		}
	}
}

File: db4oj/core/src/com/db4o/inside/query/processor/QESmaller.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o.inside.query.processor;



/**
 * @exclude
 */
public class QESmaller extends QEAbstract
{
	boolean evaluate(QConObject a_constraint, QCandidate a_candidate, Object a_value){
		if(a_value == null){
			return false;
		}
		return a_constraint.getComparator(a_candidate).isSmaller(a_value);
	}
	
	public void indexBitMap(boolean[] bits){
	    bits[QE.SMALLER] = true;
	}
	
}
